file.go

```diff
@@ -112,8 +121,99 @@ func extractFiles(zipFilePath, filesToExtractFromZipPath, localPath string) erro
 	return nil
 }
 
+// Returns true if the given file is a symlink to a file or dir
+func IsSymLink(f *zip.File) bool {
+	return f.FileInfo().Mode() & os.ModeSymlink == os.ModeSymlink
+}
+
+// Read the contents of a ZIP archive file as a byte array
+func readFileFromZip(f *zip.File) ([]byte, error) {
+	readCloser, err := f.Open()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to open file %s: %s", f.Name, err)
+	}
+
+	byteArray, err := ioutil.ReadAll(readCloser)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read file %s: %s", f.Name, err)
+	}
+
+	return byteArray, nil
+}
+
+// Given a *zip.File (a file located in a ZIP archive), write the the file to the given path
+func writeFileFromZip(f *zip.File, path string) error {
```

Comment:
When are we dealing with zip files?

Reply:
The way Fetch works is by downloading a Zip file from a repo, and then writing only the applicable files. When you extract a Zip file with Go, Go just returns a `[]*zip.File`, and you can choose to write these files to disk or not.

So this `writeFileFromZip` function is taking one of those files and writing it to disk. BTW, this functionality was there before. I just opted to encapsulate it in a function.